addons/survey/models/survey_user_input.py

```diff
@@ -591,10 +636,35 @@ class SurveyUserInputLine(models.Model):
     value_text_box = fields.Text('Free Text answer')
     suggested_answer_id = fields.Many2one('survey.question.answer', string="Suggested answer")
     matrix_row_id = fields.Many2one('survey.question.answer', string="Row answer")
+    answer_display_name = fields.Char('Answer', compute="_compute_answer_display_name",
+        help="Helper field that shows the user's answer as readable text for any type of question")
     # scoring
     answer_score = fields.Float('Score')
     answer_is_correct = fields.Boolean('Correct')
 
+    @api.depends('answer_type')
+    def _compute_answer_display_name(self):
+        for line in self:
+            if line.answer_type == 'char_box':
+                line.answer_display_name = line.value_char_box
+            elif line.answer_type == 'text_box':
+                line.answer_display_name = line.value_text_box[:50]
```

Comment:
(just to be traceback proof)
```suggestion
                line.answer_display_name = line.value_text_box and line.value_text_box[:50]
```